module-services/service-desktop/Sync.cpp

```diff
@@ -0,0 +1,205 @@
+﻿// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include <service-desktop/Sync.hpp>
+#include <service-db/DBServiceAPI.hpp>
+#include <purefs/filesystem_paths.hpp>
+
+#include <microtar.hpp>
+
+namespace sys
+{
+    class Service;
+} // namespace sys
+
+static bool isValidDirentry(const std::filesystem::directory_entry &direntry)
+{
+    return direntry.path() != "." && direntry.path() != ".." && direntry.path() != "...";
+}
+
+Sync::CompletionCode Sync::PrepareSyncPackage(sys::Service *ownerService, std::filesystem::path &path)
+{
+    assert(ownerService != nullptr);
+    LOG_INFO("Sync package preparation started...");
+
+    if (!Sync::RemoveSyncDir(path)) {
+        return CompletionCode::FSError;
+    }
+
+    if (!Sync::CreateSyncDir(path)) {
+        return CompletionCode::FSError;
+    }
+
+    LOG_DEBUG("Sync package preparation started...");
+    if (!DBServiceAPI::DBPrepareSyncPackage(ownerService, path)) {
+        LOG_ERROR("Sync package preparation, quitting...");
+        Sync::RemoveSyncDir(path);
+        return CompletionCode::DBError;
+    }
+
+    LOG_DEBUG("Packing files");
+    if (!Sync::PackSyncFiles(path)) {
+        LOG_ERROR("Failed pack sync files");
+        Sync::RemoveSyncDir(path);
```

Comment:
Sync dir should be removed only in case of these two errors, not always in PrepareSyncPackage() function.
Not sure if using gsl::finally will help here.

Reply:
OK, agreed. It looked like a missing call to `RemoveSyncDir` below for me. :)

Reply:
We can't because the tar file is in this directory :)
It needs to be downloaded by Mudita Center first.
After the file is downloaded, then MC requests to delete this directory using the filesystem endpoint.